src/Generator/Generators/C/CppTypePrinter.cs

```diff
@@ -522,11 +523,23 @@ public override TypePrinterResult VisitMethodDecl(Method method)
             var functionType = method.FunctionType.Type.Desugar() as FunctionType;
             if (functionType == null)
                 return string.Empty;
-            var returnType = method.IsConstructor || method.IsDestructor ||
+            var @return = method.IsConstructor || method.IsDestructor ||
                 method.OperatorKind == CXXOperatorKind.Conversion ||
                 method.OperatorKind == CXXOperatorKind.ExplicitConversion ?
-                string.Empty : $"{method.OriginalReturnType.Visit(this)} ";
-            var @class = method.Namespace.Visit(this);
+                new TypePrinterResult() : method.OriginalReturnType.Visit(this);
+            string @class;
+            switch (MethodScopeKind)
+            {
+                case TypePrintScopeKind.Qualified:
+                    @class = $"{method.Namespace.Visit(this)}::";
+                    break;
+                case TypePrintScopeKind.GlobalQualified:
+                    @class = $"::{method.Namespace.Visit(this)}::";
+                    break;
+                default:
+                    @class = string.Empty;
+                    break;
+            }
```

Comment:
I think we should move this to a static method otherwise `VisitMethodDecl` will keep getting bigger. We could do the same for the `noexcept` specifier.

Reply:
I agree it's neater, done.